src/engine/resolvers.rs

```diff
@@ -80,6 +81,65 @@ where
         }
     }
 
+    /// Returns the resolver input deserialized into a structure of type T that
+    /// implements the serde `Deserialize` trait.
+    ///
+    /// # Errors
+    ///
+    /// Returns an [`Error]` variant [`InputNotFound`] if no input field was passed
+    /// to the query, [`TypeConversionFailed`] if unable to convert a warpgrapher Value
+    /// to a serde_json Value, and [`JsonDeserializationFaild`] if unable to parse the
+    /// input data into a struct of type T.
+    ///
+    /// [`Error`]: ../../error/enum.Error.html
+    /// [`InputNotFound`]: ../../error/enum.Error.html#variant.InputNotFound
+    /// [`TypeConversionFailed`]: ../../error/enum.Error.html#variant.TypeConversionFailed
+    /// [`JsonDeserializationFailed`]: ../../error/enum.Error.html#variant.JsonDeserializationFailed
+    pub fn parse_input<T: serde::de::DeserializeOwned>(&self) -> Result<T, Error> {
```

Comment:
I'd recommend naming the function `input`, as it returns the input, but dropping parse_, as there's not really any parsing that happens in the function.

Reply:
Fixed.